test/adversary/transpile-adversary.mjs

```diff
@@ -111,7 +111,7 @@ const transpileBlockBuilder = (_pathToSrc, _pathToInject) => {
       proposer,
       transactionHashes: transactions.map(t => t.transactionHash),
       transactionHashesRoot: this.calcTransactionHashesRoot(transactions),
-      leafCount: badBlock.leafCount,
+      leafCount: timber.leafCount,
```

Comment:
I think this still needs to be `badBlock` - because that we may make a challenge with `incorrectLeafCount`?

Reply:
i think you're right

Reply:
@IlyasRidhuan should i remove the `getTreeByBlockNumberL2` update?

Reply:
No I think it's better this way. `getTreeByBlockNumberL2` is idempotent so I feel better about it